ngrinder-controller/src/main/java/org/ngrinder/script/model/GitApi.java

```diff
@@ -0,0 +1,39 @@
+/* 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. 
+ */
+package org.ngrinder.script.model;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang.ArrayUtils.addAll;
+
+/**
+ * Support for Git Api
+ * 
+ * @since 3.5.0
+ */
+public enum GitApi {
+	REPO("https://%s/repos/%s/%s"),
+	REF("https://%s/repos/%s/%s/git/ref/heads/%s"),
+	TREES("https://%s/repos/%s/%s/git/trees/%s?recursive=1");
+
+	private String apiTemplate;
+
+	GitApi(String apiTemplate) {
+		this.apiTemplate = apiTemplate;
+	}
+
+	public String buildUrl(GitConfig gitConfig, Object... args) {
+		String[] baseArgs = { gitConfig.getBaseUrl(), gitConfig.getOwner(), gitConfig.getRepo() };
+		return format(apiTemplate, addAll(baseArgs, args));
+	}
+}
```

Comment:
Why don't you use github api library like this.
https://github-api.kohsuke.org/

Reply:
I modified to use this.